backend/src/api/endpoints/document_tag.py

```diff
@@ -27,8 +29,22 @@
     description="Creates a new DocumentTag and returns it with the generated ID.",
 )
 async def create_new_doc_tag(
-    *, db: Session = Depends(get_db_session), doc_tag: DocumentTagCreate
+    *,
+    db: Session = Depends(get_db_session),
+    doc_tag: DocumentTagCreate,
+    authz_user: AuthzUser = Depends(),
 ) -> DocumentTagRead:
+    authz_user.assert_in_project(doc_tag.project_id)
+
+    if doc_tag.parent_tag_id is not None and doc_tag.parent_tag_id != -1:
+        authz_user.assert_in_same_project_as(Crud.DOCUMENT_TAG, doc_tag.parent_tag_id)
+
+        parent_tag = crud_document_tag.read(db, doc_tag.parent_tag_id)
+        authz_user.assert_bool(
```

Comment:
Dafür könnte man eine generelle Methode schreiben: `assert_objects_in_the_same_project`. Die kann man dann auch immer verwenden, wenns zB um Tagging geht oder Memo create etc. (siehe andere Kommentare bzgl `assert_objects_in_the_same_project` )

Reply:
Done! Aktuell funktioniert sie allerdings nicht für objekte, die noch nicht in der datenbank stehen, da `get_parent_project_id` das nicht kann.